src/main/java/network/brightspots/rcv/CandidatesAtRanking.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * RCTab
+ * Copyright (c) 2017-2022 Bright Spots Developers.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Purpose: Low-Memory container for a list of candidates at a single ranking.
+ * Design: Optimized for the case of one candidate per ranking.
+ * Conditions: Always.
+ * Version history: see https://github.com/BrightSpots/rcv.
+ */
+
+
+package network.brightspots.rcv;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+class CandidatesAtRanking implements Iterable<String> {
+  static class CandidatesAtRankingIterator implements Iterator<String> {
+    private int iteratorIndex = 0;
+
+    private CandidatesAtRanking candidates;
+
+    CandidatesAtRankingIterator(CandidatesAtRanking candidates) {
+      this.candidates = candidates;
+    }
+
+    public boolean hasNext() {
+      return iteratorIndex < candidates.candidateNames.length;
+    }
+
+    public String next() {
+      if (iteratorIndex == candidates.candidateNames.length) {
+        throw new NoSuchElementException();
+      }
+
+      iteratorIndex++;
+
+      return candidates.get(iteratorIndex - 1);
+    }
+
+    public void remove() {
+      throw new UnsupportedOperationException();
+    }
+  }
+
+  private String[] candidateNames;
+
+  int count() {
+    int numCandidates = 0;
+    if (candidateNames != null) {
+      numCandidates = candidateNames.length;
+    }
+    return numCandidates;
```

Comment:
to be concise:
```
return candidateNames == null ? 0 : candidateNames.length;
```

Reply:
I wasn't sure if trinary operaters counted as multiple returns -- I did some investigation and the bytecode looked [sufficiently close](https://stackoverflow.com/a/63045843/1057105) to `if/else`. Further, the exact implementation isn't specified, so it can vary from implementation to implementation.

All this to ask: what are the exact rules about multiple returns, and are we sure trinary operators follow these rules?

Reply:
I really don't think the certification folks are looking at the bytecode. :) I think they just want to see that every function has at most one `return` statement and, if present, it's the final statement. So a ternary should be fine, as far as I know.